test/buffy/throttle_and_timed_test.exs

```diff
@@ -0,0 +1,251 @@
+      test_pid = self()
+      MyTimedThrottler.throttle(%{test_pid: test_pid, prev: prev})
+
+      # Initial throttle is 10 msec so should receive within 20 msec
+      assert_receive {:ok, %{prev: ^prev}, now}, 200
+      assert System.convert_time_unit(now - prev, :native, :millisecond) < 20
+
+      # Inbox timeout triggers at 100 msec so should receive "around" that time
+      assert_receive {:ok, %{prev: ^prev}, now2}, 200
+      diff = System.convert_time_unit(now2 - now, :native, :millisecond)
+      assert :erlang.abs(diff - 100) < 10
+
+      # Confirm another inbox timeout triggered
+      assert_receive {:ok, %{prev: ^prev}, now3}, 200
+      diff = System.convert_time_unit(now3 - now2, :native, :millisecond)
+      assert :erlang.abs(diff - 100) < 10
+    end
+
+    test "should throttle all incoming triggers when work is already scheduled" do
+      DynamicSupervisor.count_children(MyDynamicSupervisor)
+      test_pid = self()
+      # trigger throttle
+      MyTimedSlowThrottler.throttle(%{test_pid: test_pid})
+
+      # trigger more throttle
+      for _ <- 1..10 do
+        Task.async(fn ->
+          MyTimedSlowThrottler.throttle(%{test_pid: test_pid})
+        end)
+      end
+
+      # assert throttled work done
+      assert_receive {:ok, _, now}, 200
+
+      # refute any other work was done
+      refute_receive {:ok, _, _now}, 200
+
+      # check inbox timeout triggered
+      assert_receive {:ok, _, now2}, 400
+      diff = System.convert_time_unit(now2 - now, :native, :millisecond)
+      assert :erlang.abs(diff - 300) < 10
+    end
+
+    test "should reset inbox timeout if throttle request comes in" do
+      test_pid = self()
+      # trigger initial throttle
+      MyTimedSlowThrottler.throttle(%{test_pid: test_pid})
+
+      # assert throttled work done
+      assert_receive {:ok, _, now}, 200
+
+      # now in inbox timeout waiting period and work scheduled via inbox timeout logic
+      # trigger a throttle
+      MyTimedSlowThrottler.throttle(%{test_pid: test_pid})
+
+      # assert the trigger happend within the throttle interval and not the inbox timeout loop interval
+      assert_receive {:ok, _, now2}, 400
+      diff = System.convert_time_unit(now2 - now, :native, :millisecond)
+      assert :erlang.abs(diff - 100) < 10
+    end
+  end
+
+  describe "handle_throttle/1" do
+    setup do
+      start_supervised!({MyDynamicSupervisor, []})
+      :ok
+    end
+
+    test "calls handle_throttle/1" do
+      check all args <- StreamData.term() do
+        assert :ok = MyZeroThrottler.throttle(%{args: args, test_pid: self()})
+        assert_receive {:ok, _, _}
+      end
+    end
+
+    test "throttles handle_throttle/1" do
+      test_pid = self()
+      for _ <- 1..200, do: MySlowThrottler.throttle(%{test_pid: test_pid})
+      assert_receive {:ok, _, _}, 200
+      refute_receive {:ok, _, _}, 200
+    end
+  end
+
+  describe ":telemetry" do
+    setup do
+      _ref =
```

Comment:
Any reason to keep this if we're not using it?

```suggestion
```

Reply:
Done in adbaea0